app/controllers/orders_controller.rb

```diff
@@ -13,56 +14,110 @@ def index
   def show
     logger.tagged("#{current_user.urs_uid} #{controller_name}_controller") do
       @order = Echo::Order.new(client: echo_client, echo_provider_token: echo_provider_token, guid: params['id'])
-
       add_breadcrumb @order.guid, order_path(@order.guid)
-
       render :show
     end
   end
 
   def search
-    logger.tagged("#{current_user.urs_uid} #{controller_name}_controller") do
-      @orders = Echo::Orders.new(client: echo_client, echo_provider_token: echo_provider_token, guids: determine_order_guids).orders
+    Rails.logger.info("starting request - #{request.uuid} timeout=#{echo_client.timeout}")
+    init_time_tracking_variables
+    guids = determine_order_guids
 
-      # if user_id param is supplied and we're not searching by guid, filter orders by given user_id
-      if params['order_guid'].blank? && params['user_id'].present?
-        @orders.select! { |order| order.owner == params['user_id'] }
-      end
+    time "search #{guids.class == Array ? guids.count : 1} orders" do
+      logger.tagged("#{current_user.urs_uid} #{controller_name}_controller") do
+        time 'time to retrieve details infos' do
+          echo_client.timeout = time_left
+          @orders = Echo::Orders.new(client: echo_client, echo_provider_token: echo_provider_token, guids: guids).orders
+        end
+
+        time 'time to precache owner guids' do
+          precache_owner_guids
+        end
+
+        # if user_id param is supplied and we're not searching by guid, filter orders by given user_id
+        if params['order_guid'].blank? && params['user_id'].present?
+          @orders.select! {|order| order.owner == params['user_id']}
+        end
 
-      @orders.sort_by!(&:created_date)
+        @orders.sort_by!(&:created_date)
+        render :search
+      end
 
-      render :search
     end
+  rescue Faraday::Error::TimeoutError
+    flash.now[:alert] = 'The order request timed out retrieving results.  Limit your search criteria and try again.'
+    render :index
   end
 
   private
 
+  # sets up initial values to track time spent issuing faraday requests.
+  # initial budget of time allowed to complete request is 270 seconds (300-30 see below)
+  # echo_client.timeout as of 5/16/19 is 300 seconds, subtracting 30 seconds for any potential processing,
```

Comment:
I'm not sure if the date is necessary, but saying why it is 300s might be helpful

Reply:
Agree and done.